### ISSUE TITLE
Can we get in contact?

### ISSUE THREAD

**Issue by joostiphone** [zearp/OptiHack#43]
Hi @zearp 

My apologies for contacting you like this. I'm trying to get in contact with you to discuss taking part in the (new) [HackinDROM app](https://www.tonymacx86.com/threads/hackindrom-app-testing-not-website.311144/). What is the best way we can chat? E.g., are you available on tonymacx86.com? 

Thanks!
Joost
[https://www.tonymacx86.com/members/joostiphone.1233058/](https://www.tonymacx86.com/members/joostiphone.1233058/ )

**Comment by zearp** [comment]
Is there a Github repo for that app or is it closed source?

**Comment by joostiphone** [comment]
There will be a GitHub page eventually, user inqnuam is the developer. The idea is to have a admin/owner per build (OpenCore EFI), so that all other users can easily upgrade their EFI with HackinDROM. We've been beta testing for quite a while now and are basically ready to onboard other build, such as the 9020 optiplex. Since you're deep knowledge on this build, I would like you to be that person for this build specifically. 

You can download the app from  https://hackindrom.zapto.org/ (below on that page). You can watch a video I made on how to update the efi with this app, perhaps this gives you a nice picture of what the app does. https://youtu.be/xRuerrG-lAU

**Comment by zearp** [comment]
I'm not sure what you need me for, I mean you can add the repo to the app and have it do what people manually do to update/upgrade. I don't really understand what my role would be. I get it's an updater app but I don't see how repo owners fit in.

Anyone using my repo can update/upgrade simply by moving their serials/etc to a text file, delete the EFI folder and replace it with te latest version found here and copy their serials/etc back. Pretty straight forward and simple. Any changes the user made they re-apply to the new EFI, they know which changes they made.

**Comment by joostiphone** [comment]
I don't mind uploading your efi to the app, instead of you. As long as you allow me to do that for this build (using your efi), that would be great. We just want to credit the person maintaining the efi and perhaps you would like this doing yourself. 

I myself am using the 9020 USFF and I think I will handle that build myself. 

Thanks for your replies so far!

**Comment by zearp** [comment]
A big issue I can see happening is people updating from older OpenCore versions that have missing and/or depreciated config entries. You'd have to know the correct values of any missing entries for each target the app supports. Even with one target this will lead to many possibilities between versions.

For example if someone upgrades from OpenCore 0.6.1 to 0.6.7 there will be a lot of changes in the config. Entries renamed, removed and added. The possibilities exponentially grow depending on the changes. Not to mention completely new additional like EC sensor support that was recently added to VirtualSMC.

For users of my EFI it would be a lot easier and less failure prone to just delete the old EFI and replace it with the latest EFI from here and copy their serials back. No need to mess about with anything. But I get the idea behind the app. I just hope it will be Open Source. If not I don't want to participate and would prefer to be excluded. I also don't want people using the tools ending up here opening issues cuz their config got messed up.

I'm a big fan of teaching people how to fish, not giving out the fish. One of the goals of my repo's is to educate myself and others and provide a stable EFI that will work and can serve as base for a more customised build. Like anyone using dGPU will need to make changes as outlined in the readme. Or people who've added an Airport card.

I don't know how this tool would handle those cases. Will it detect dGPU and try to enable it by removing entries from the config? Will it detect an Airport card and add the relevant device settings to the config? So many things to consider even for this EFI that uses a prebuild system. I can't imagine  the sheer number of different configurations for any system not based on a prebuild.

You can use the EFI in the app but only if the app will be Open Source and of course give credits. I don't wanna be part of a new closed source "beast tool". I don't really like apps that keep the end user in the dark as to whats going on. It prevents people from learning how their hack works and therefor limit their ability to maintain it. They will become dependant on (closed) tools to keep their setup working. Such dependance isn't something we should strive for in the community. But thats just my opinion.

**Comment by zearp** [comment]
I've noticed lots of progress in the threads about this, but haven't seen any sources being published. By the looks of if it this app wants to be some kind of one-click solution that doesn't really teach the user anything and makes them dependant on the app. There's also something to say about using other people's work in the app without credit and just take things from Github repo's and present it as your own work. Unethical and possibly illegal. I noticed the 9020 build is credited to you while you just took my EFI.

I guess this is why the open source community doesn't really like Tonymac. Developers of the [Dortania](https://github.com/khronokernel/Tonymacx86-stance) guide are, developers of kexts such as [IntelMausi](https://github.com/Mieze/IntelMausiEthernet/blob/master/IntelMausiEthernet/IntelMausiSetup.cpp#L224-L226) are. Many more are too. And they all have very good reasons. I wasn't even aware of all that but I am now. Plus I saw mods on Tonymac recommend using tools such a OpenCore Configurator. Something the developers of OpenCore highly discourage as it can easily break things and teaches the user nothing about OpenCore or its workings.

I'm curious if there will ever be a source for HackinDROM on Github as you alluded to. Till that time I no longer want my work to be used in this closed source app, especially since it has some kind of weird admin access to push EFI's to the back-end of the app. Something that could easily be abused by a bad actor that pushes an EFI with backdoors to a bunch of people for example.

I really don't understand why there's so much time and effort put into an app that gives out fish. Why not put that time into something that teaches people how to fish? Or why not put in time to contribute to existing projects and make them better? There are things to test and bugs to smash in OpenCore and related kexts. Why not put time and effort in that? Make things better for everybody. But Tonymac just takes and doesn't give back to the community they take everything from. The whole app would be useless if it wasn't for open source development.

Oh one more thing I noticed when reading Tonymac's TOS. Users are not allowed to post links with referral codes. Understandable. But at the same time any link a user posts to Amazon or Newegg automatically gets referral codes inserted for Tonymac. That's not very ethical is it? It's one thing if links posted by admins or mods have them added. But to add them automatically whenever someone posts any links to those sites just screams *we wanna make money* to me. Something the Hackintosh community is not about.

Another thing I noticed is that whenever someone links to my repo's on Tonymac those posts are deleted. It has happened quite a few times that I see a referral in my traffic stats leading to a Tonymac thread only to not find any post linking to me. Sometimes Google's cache still has the thread cached with the original post in there. A quick check on that users profile often leads to a banned profile. So it seems some mods over there are banning people who link to Github repo's. Pretty sad.

Speaking of banned users, so many posts on Reddit and other Hackintosh forums talk about how easy it is to get banned there. People getting banned for frivolous reasons or no reason at all. The more I look into Tonymac the more it looks like. cult and not a healthy community with open discussions and debate. Anyone going against the grain will be banned. Gotta keep the echo chamber in tact I guess.

tl;dr Please don't use any of my work for this app, not until the sources are public.

**Comment by joostiphone** [comment]
I'm sorry, but I didn't use your EFI files. I followed the tutorial on Tony, where I'm referring to. I noticed your build on Github finding an admin for more builds, which you respectfully declined. Your build is also not the same as mine.

I'm also not crediting myself. I'm actually crediting trs96 for his instructions. I did however put the EFI together myself and shared that through the app. I would motivate you to not putting conclusions that fast, and just ask upfront if you would've liked clarification on this. Still, that's completely up to you to decide.

In all transparency I approached you because personally, I would like to see as many builds in that app as possible, coming from the EFI maintainers themselves. I respect that you don't want to participate in this and I therefore never used your work (also not personally). I also, therefore, didn't proceed to reply, because it's clear to me you don't want to be involved, which is fine by me also. 

tl;dr I'm not using your work, at all.

**Comment by joostiphone** [comment]
I also just want to add for clarification purposes, I'm a user of the Tony forum, not an admin/owner or such. So I take your comments/opinions on that forum for what it is. I have another opinion though, as a lot of people over there are to help and I see many great discussions. I'm still not banned. 

I don't have a saying in HackinDROM becoming open-sourced (I'm not a dev or something), but I understood that this is the plan eventually. I personally still like the idea of the app which is in my opinion unburden other Hackintosh users.

**Comment by zearp** [comment]
I see I wasted my time writing a response. I'll remove this issue and hope to see my work removed from this app. You're clearly not interested in any of my arguments, or arguments form others that contribute a lot more than me. Guess it further explains why so many developers are anti Tonymac. But I guess it's all their fault and Tonymac does everything right. Truly a cult.
